### PR TITLE
Handle re-claim while the agent is running in new architecture

### PR DESCRIPTION
**Pull request by MrZammler**
##### Summary

This PR adds some modifications that allow the agent to handle a re-claim while the agent is running for the new architecture. In this scenario, the agent will disconnect, send an `UpdateAgentConnection` which has the old `claim_id`, and then another `UpdateAgentConnection` containing the new `claim_id`. In cloud terms, the old node instance will become unreachable, and the new node instance will start updating.

Without this PR the agent requires a restart to re-appear in the cloud, and if it doesn't restart the cloud will have a node instance which does not update any more, and a new one in unreachable state.

There is also a small change in the `netdata-claim.sh.in` script: If the cloud responds with `ErrAlreadyClaimed` do not overwrite the the `claimed_id` file. This helps with the following scenario (but we can discuss if I'm missing something):

Agent is claimed for the first time without the `-id` parameter. `claimed_id` contains the `machine_guid`
Agent is re-claimed with the `-id` parameter: `claimed_id` now contains a new id.
Agent is re-claimed without the `-id` parameter. `claimed_id` is overwritten by the `machine_guid`.

##### Test Plan

In a whitelisted space, claim a fresh instance of an agent. Observe that it starts working.
Re-claim the agent using the `-id` parameter.
The old agent instance should become unreachable, and the new one will start updating.

##### Additional Information

There is really more to this than this PR. If re-claimed, the agent appears to the cloud as a completely new instance. As such, the cloud will request streaming from sequence id 1, which might not be available by then to the agent (db rotated). Snapshot events for alerts, and re-syncs for charts should fix the situation (but this is a general problem with re-claiming).

### CODE REVIEW

File: aclk/aclk.c [hunk 1]
```diff
@@ -320,6 +320,15 @@ static int handle_connection(mqtt_wss_client client)
             return 1;
         }
 
+        if (aclk_kill_link) { // User has reloaded the claiming state
```
Comment (underhood) [inline aclk/aclk.c:323]: can we change it to :
`if (disconnect_req || aclk_kill_link) {`
to not duplicate code?

File: aclk/aclk_rrdhost_state.h [hunk 2]
```diff
@@ -30,6 +30,7 @@ typedef enum aclk_agent_state {
 
 typedef struct aclk_rrdhost_state {
     char *claimed_id; // Claimed ID if host has one otherwise NULL
+    char *prev_claimed_id; // Claimed ID if changed (reclaimed) during runtime
```
Comment (underhood) [inline aclk/aclk_rrdhost_state.h:33]: clear it on shutdown?
perhaps in `rrdhost_free` ?

File: aclk/aclk.c (hunk 1 above)
Comment (MrZammler) [reply aclk/aclk.c:323]: Yes @underhood , you're right. Will change it. Had it on different block for some checks, it doesn't need to be in it's own block.

File: aclk/aclk_rrdhost_state.h (hunk 2 above)
Comment (MrZammler) [reply aclk/aclk_rrdhost_state.h:33]: Correct! Pushed, it but leaving the converstation un-resolved to check.

**Review by siamaktavakoli — APPROVED** [review]
I tested, and it worked as it was expected.

**Review by underhood — APPROVED** [review]

**Review by stelfrag — APPROVED** [review]

**Comment by MrZammler** [comment]
Thanks!!

**Pull request merged**